src/Raven.Server/Documents/Sharding/ShardedDocumentDatabase.cs

```diff
@@ -215,30 +222,73 @@ public async Task DeleteBucketAsync(int bucket, long migrationIndex, long confir
 
+        var delay = TimeSpan.FromSeconds(1);
         while (true)
         {
             var cmd = new DeleteBucketCommand(this, bucket, uptoChangeVector);
-            await TxMerger.Enqueue(cmd);
+            try
+            {
+                await TxMerger.Enqueue(cmd);
+            }
+            catch (Exception exception)
+            {
+                // if an error occurs during the execution of DeleteBucketCommand,
+                // we need to handle it gracefully to avoid rapid retry attempts
+
+                RaiseNotificationOnDeleteBucketFailure(bucket, exception);
+
+                if (delay >= TimeSpan.FromMinutes(5))
+                {
+                    // if the delay exceeds the maximum timeout, throw an exception and break the loop
+                    await Task.FromException(exception);
+                    return;
+                }
+
+                await Task.Delay(delay, cancellationToken: DatabaseShutdown);
+                delay = delay.Multiply(2);
+                continue;
+            }
 
             switch (cmd.Result)
             {
                 // no documents in the bucket / everything was deleted
                 case DeleteBucketCommand.DeleteBucketResult.Empty:
                     await ServerStore.Sharding.SourceMigrationCleanup(ShardedDatabaseName, bucket, migrationIndex);
+                    DismissNotificationOnDeleteBucketSuccessIfNeeded();
                     return;
                 // some documents skipped and left in the bucket
                 case DeleteBucketCommand.DeleteBucketResult.Skipped:
                     return;
                 // we have more docs, batch limit reached.
                 case DeleteBucketCommand.DeleteBucketResult.FullBatch:
                 case DeleteBucketCommand.DeleteBucketResult.ReachedTransactionLimit:
+                    delay = TimeSpan.FromSeconds(1);
                     continue;
                 default:
                     throw new ArgumentOutOfRangeException();
             }
         }
     }
 
+    private void RaiseNotificationOnDeleteBucketFailure(int bucket, Exception exception)
+    {
+        var msg = $"An error occurred while attempting to clean up bucket '{bucket}' from source shard '{ShardNumber}' [{Name}].";
+
+        if (_logger.IsInfoEnabled)
+            _logger.Info(msg, exception);
+
+        ServerStore.NotificationCenter.Add(AlertRaised.Create(
+            ShardedDatabaseName,
+            "Resharding Delay Due to an Error",
+            msg,
+            AlertType.ClusterTransactionFailure,
+            NotificationSeverity.Error,
+            details: new ExceptionDetails(exception),
+            key: $"{Name}/ReshardingFailure"));
+    }
+
+    private void DismissNotificationOnDeleteBucketSuccessIfNeeded() => ServerStore.NotificationCenter.Dismiss($"AlertRaised/ClusterTransactionFailure/{Name}/ReshardingFailure", sendNotificationEvenIfDoesntExist: false);
```

Comment:
use `AlertRaised.GetKey()` and make `key` a property (and also use it in line 287)

Reply:
done